docker/build/devdeps.Dockerfile

```diff
@@ -135,7 +135,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     && python3 -m pip install --no-cache-dir \
         lit pytest numpy \
         fastapi uvicorn pydantic llvmlite \
-        openfermionpyscf \
+        scipy==1.10.1 openfermionpyscf==0.5 \
```

Comment:
It might worth to pin the version for the other packages as well. 

Also, couldn't we factor these in a `requirements.txt` and have pip install from that file? Seems like this is something we would need also when building outside a docker image, e.g. have a `requirements.txt` in the repository and have the docker use that file.

Reply:
Agreed. It probably makes sense to tackle that as part of larger revision of our python installation in https://github.com/NVIDIA/cuda-quantum/pull/304. I only did the minimal change here to restore the CI.